pyvcloud/vcd/org.py

```diff
@@ -118,6 +118,27 @@ def get_catalog(self, name):
                 return self.client.get_resource(link.href)
         raise Exception('Catalog not found.')
 
+    def get_user(self, user_name):
+        if self.resource is None:
+            self.resource = self.client.get_resource(self.href)
+        resource_type = 'adminUser' if self.is_admin else 'user'
+        org_filter = 'org==%s' % self.resource.get('href') if self.is_admin else None
+        result = []
+        q = self.client.get_typed_query(
+            resource_type,
+            query_result_format=QueryResultFormat.REFERENCES,
+            equality_filter=('name', user_name),
+            qfilter=org_filter
+            )
+        records = list(q.execute())
+        if len(records) == 0:
+            raise Exception('user not found')
+        elif len(records) > 1:
```

Comment:
It can catch our own code bugs.
For example : When this method is invoked by Sys Admin to get an user with the same name in multiple organizations, it has to return unique result.

Reply:
I don't think the change_ower command should fail when the same user is present two orgs(Even when invoked by sysadmin). I think we should restrict the user search to the 'org_in_use' even when invoked by sysadmin.

What if we had 3 org
system
org1 with userA1, userB1 with vapp1
org2 with userA2, userB2, userA1 with vaap2

We login as sysadmin
switch to org1

1. invoke change_owner(userA2, vaap1)
Will it now set userA2 as owner of vapp1 ? (userA2 is not a user in org1 and should fail)

2. invoke change_owner(userA1, vaap1)
   Should we fail this command since now get_user() will return 2 users and throw an exception? I don't think we should fail in this case. There could be common username between many orgs and this should not be the reason to fail the command.

I think we should have org_filter to the current 'org_in_use' for both sysadmin and nonadmin case.

Reply:
Anusuya, 

1. invoke change_owner(userA2, vapp1) - this will fail.
2. invoke change_owner(userA1, vapp1) - this will not fail and will correctly set the user with that of org-in-use.

org_filter is always set to org_in_use. Refer to vcd-cli/vapp/change_owner method.
_______
My point earlier was if ever in future, developer makes a mistake in passing the incorrect values from vcd-cli, the above check for more than one record will catch it.

Reply:
Sorry for not being clear in my earlier comment.